tests/uefistored/test_cert_inheritance.py

```diff
@@ -325,15 +326,17 @@ def test_set_pool_certificates_partial(self, host):
                 check_disk_cert_md5sum(h, key, pool_auths[key].auth)
             assert not h.file_exists(f'{CERT_DIR}/{missing_key}.auth')
 
-    def test_clear_certificates_from_pool(self, host):
-        keys = ['PK', 'KEK', 'db', 'dbx']
-        pool_auths = generate_keys(as_dict=True)
-        host.pool.install_custom_uefi_certs([pool_auths[key] for key in keys])
-        host.pool.clear_uefi_certs()
+    @pytest.mark.small_vm
+    def test_clear_certificates_from_pool(self, host, uefi_vm):
+        vm = uefi_vm
```

Comment:
Now your test, when executed alone, won't necessarily have a VM clean of certificates. It's the `uefi_vm_and_snapshot` fixture that cleans them in its setup, not the `uefi_vm` fixture.

Plus, using `uefi_vm_and_snapshot` would let you revert the VM at the end of this test, for other tests that may be executed afterwards (pytest's execution order is not always that of the file, or we may add more tests in the future) and that use the same VM.